driver-core/src/main/java/com/datastax/driver/core/policies/HostFilterPolicy.java

```diff
@@ -0,0 +1,226 @@
+        // Just delegate to the child policy, since we filter the hosts not white
+        // listed upfront, the child policy will never see a host that is not white
+        // listed and thus can't return one.
+        return childPolicy.newQueryPlan(loggedKeyspace, statement);
+    }
+
+    @Override
+    public void onUp(Host host) {
+        if (predicate.apply(host))
+            childPolicy.onUp(host);
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public void onSuspected(Host host) {
+        if (predicate.apply(host))
+            childPolicy.onSuspected(host);
+    }
+
+    @Override
+    public void onDown(Host host) {
+        if (predicate.apply(host))
+            childPolicy.onDown(host);
+    }
+
+    @Override
+    public void onAdd(Host host) {
+        if (predicate.apply(host))
+            childPolicy.onAdd(host);
+    }
+
+    @Override
+    public void onRemove(Host host) {
+        if (predicate.apply(host))
+            childPolicy.onRemove(host);
+    }
+
+    @Override
+    public void close() {
+        if (childPolicy instanceof CloseableLoadBalancingPolicy)
+            ((CloseableLoadBalancingPolicy)childPolicy).close();
+    }
+    
+    /**
+     * Create a new policy that wraps the provided child policy but only "allow/forbids" hosts
+     * from the provided list.
+     *
+     * @param childPolicy the wrapped policy.
+     * @param hosts the hosts.
+     * @param allow when true allow hosts from the provided list to be connected to 
+     * (whether they will get connected to or not depends on the child policy), otherwise forbids these hosts
+     */
+    public static HostFilterPolicy mkHostAddressListPolicy( LoadBalancingPolicy childPolicy, Collection<InetSocketAddress> hosts, boolean allow) {
+        Predicate<Host> predicate = mkHostAddressPredicate(hosts);
+        return new HostFilterPolicy(childPolicy, allow ? predicate : Predicates.not(predicate));
+    }
+    
+    /**
+     * Create a new policy that wraps the provided child policy but only "allow/forbids" hosts
+     * whose DC belongs to the provided list.
+     *
+     * @param childPolicy the wrapped policy.
+     * @param dcs the DCs.
+     * @param allow when true allow hosts whose DC from the provided list to be connected to 
+     * (whether they will get connected to or not depends on the child policy), otherwise forbids these hosts
+     */
+    public static HostFilterPolicy mkDCListPolicy( LoadBalancingPolicy childPolicy, Collection<String> dcs, boolean allow) {
+        Predicate<Host> predicate = mkHostDCPredicate(dcs, allow);
+        return new HostFilterPolicy(childPolicy, allow ? predicate : Predicates.not(predicate));
+    }
+    
+    /**
+     * Create a predicate based on membership to a list of addresses
+     * @param hosts the host addresses
+     * @return the predicate
+     */
+    public static Predicate<Host> mkHostAddressPredicate( Collection<InetSocketAddress> hosts) {
```

Comment:
Any reason to make these public? We try to expose as little as we can to avoid future backward compatibility issues.